core/saltfilter/bloom.go

```diff
@@ -0,0 +1,34 @@
+// Package saltfilter used for reuse detection ONLY
+package saltfilter
+
+import (
+	"sync"
+
+	"github.com/willf/bloom"
+)
+
+var (
+	filterInstance *bloom.BloomFilter
+	itemCount      = 0
+	mux            = sync.Mutex{}
+)
+
+const itemSize = 1e6
+
+func init() {
+	filterInstance = bloom.NewWithEstimates(itemSize, 0.003)
```

Comment:
This value should be configurable as well. The use of Bloom filters should be made optional and configurable according to https://github.com/shadowsocks/shadowsocks-org/issues/44